### PR TITLE
Regression updates

### PR DESCRIPTION
**Pull request by amadeuspzs**
# Summary

**What/Why**

* Fixes catboost implementation
* Removes hyperparameters as per approach in classification
* Improves plots as per approach in classification

**Your approach**

* Updated notebooks

**What to review**

* 05a Modelling - regression
* 05b Modelling - regression fairness

# Checklists

This pull/merge request meets the following requirements:

- [X] code runs
- [X] [developments are ethical][data-ethics-framework]
- [X] developments are secure (no data or notebook output)
- [X] you have made proportionate checks that the code works correctly
- [X] [minimum usable documentation][agilemodeling] written in the `docs` folder

Comments have been added below around the incomplete checks.

[agilemodeling]: http://agilemodeling.com/essays/documentLate.htm
[data-ethics-framework]: https://www.gov.uk/government/publications/data-ethics-framework

### CODE REVIEW

File: notebooks/05b Modelling - regression fairness.ipynb [hunk 1]
```diff
@@ -147,10 +164,20 @@
...
    "source": [
-    "# load models from outside the git tree\n",
-    "with open(\"../../models/regression.pickle\", \"rb\") as handle:\n",
-    "    models = pickle.load(handle)\n",
-    "models"
+    "for feature in sensitive_columns:\n",
+    "    data = pd.DataFrame(columns=[\"subfeature\", \"count\"])\n",
+    "    # iterate over all subcategories for the feature\n",
+    "    for subfeature in fairness_df[feature].unique():\n",
+    "        subset = fairness_df[fairness_df[feature] == subfeature]\n",
```
Comment (matthewcooper19) [inline notebooks/05b Modelling - regression fairness.ipynb:171]: fairness_df is referenced here before it has been created

File: notebooks/05b Modelling - regression fairness.ipynb [hunk 2]
```diff
@@ -185,16 +211,20 @@
...
+    "        mean_los = subset.LENGTH_OF_STAY.mean()\n",
+    "        df = pd.DataFrame(data=[{\"subfeature\": subfeature, \"mean_los\": mean_los}])\n",
+    "        data = pd.concat([data, df])\n",
     "\n",
-    "# Create a combined data frame\n",
-    "fairness_df = X_test.copy()\n",
-    "fairness_df[\"LENGTH_OF_STAY\"] = y_test\n",
-    "fairness_df[\"LENGTH_OF_STAY_PREDICTED\"] = preds"
+    "    fig = px.bar(data, x=\"subfeature\", y=\"mean_los\", title=feature, orientation=\"v\")\n",
```
Comment (matthewcooper19) [inline notebooks/05b Modelling - regression fairness.ipynb:224]: Up to you, but could be worth considering adding a y-axis label that is more descriptive

File: notebooks/05b Modelling - regression fairness.ipynb (hunk 1 above)
Comment (amadeuspzs) [reply notebooks/05b Modelling - regression fairness.ipynb]: Apologies, this has been fixed

File: notebooks/05b Modelling - regression fairness.ipynb (hunk 2 above)
Comment (amadeuspzs) [reply notebooks/05b Modelling - regression fairness.ipynb:230]: Agreed and updated

**Review by matthewcooper19 — APPROVED** [review]

**Pull request merged**